### DISCUSSION TITLE
Peer policy

### DISCUSSION THREAD

**Discussion by aer0nix** [arvidn/libtorrent#6838] [Ideas]
uTorrent have pretty much feature:  peer policy. Looking for local peers without LSD multicast storming.

> **isp.peer_policy_enable**: _This option enables peer policy functionality, which sets weights to different IP ranges._
> **isp.peer_policy_override**: _This option overrides the peer policy._
> **isp.peer_policy_url**: _This option sets a URL to the ISP's peer policy._

example of **peer_policy_list.xml**:
```
<btpolicy>
  <iprange start="192.168.1.1" end="192.168.1.255" weight="10" />
  <iprange start="192.168.2.1" end="192.168.2.255" weight="1" />
</btpolicy>
```

Does libtorrent support it?

**Comment by arvidn** [comment]
The short answer is: "no". Libtorrent does not support this feature.

My recollection is that this feature creates an affinity to connect to peers that are either on the same ISP or on networks the ISP is peering with. The "policy" is the ISP having the ability to nudge the client to connect to peers it prefers, whose bandwidth is cheaper.

I don't believe it's about *discovering* peers.

If you have a different understanding, please provide some references.
